haystack/generator/transformers.py

```diff
@@ -18,91 +19,227 @@ class RAGGeneratorType(Enum):
 
 
 class RAGGenerator(BaseGenerator):
+    """
+        Implementation of Facebook's Retrieval-Augmented Generator (https://arxiv.org/abs/2005.11401) based on
+        HuggingFace's transformers (https://huggingface.co/transformers/model_doc/rag.html).
 
+        |  With the generator, you can:
+
+            - directly get generate predictions via predict()
+    """
     def __init__(
             self,
             model_name_or_path: str = "facebook/rag-token-nq",
+            passage_embedding_model: str = "facebook/dpr-ctx_encoder-single-nq-base",
```

Comment:
I think we should make `passage_embedding_model` optional. The default case will be to use the generator together with a DPR Retriever and in that case we can pass the passage embeddings straight to the generator. Loading the DPRContextEncoder twice will waste quite some (GPU) RAM in production.

The only use case that I see for having this arg is the usage of the Generator together with a sparse retriever (e.g. ElasticsearchRetriever) where we won't have embeddings. Am I am missing another use case?

Reply:
Yes you are correct, I will make this optional.

Currently DPR Retriever do not add `embedding` along with retrieved document. We need to modify `query_by_embedding` function of `FAISSDocumentStore` to add `embedding` with returned document when some flag (for example `attach_embedding`) is passed. wdyt?

Reply:
Yep, good point. 
I think ElasticsearchDocumentStore is already returning the embedding with the doc by default and this can be switched off by setting `ElasticsearchDocumentStore(excluded_meta_data=["embedding"])`. However, this is not really nice as we already moved `embedding` out of `doc.meta` into `doc.embedding`. 

For FAISS the behavior is different and no embedding gets returned. Let's harmonize this.
How about:
- We add an arg `return_embedding` in both DocumentStores? 
- Default is `True`

Reply:
I think let's not mix multiple improvements in the single PR. 

Let's create another task to add `return_embedding` flag in `ElasticsearchDocumentStore`, `FAISSDocumentStore` or any embedding store. In generator we already have lever to generate embeddings if required.

Reply:
Agree. Created https://github.com/deepset-ai/haystack/issues/503